### PR TITLE
add recipe for swi-prolog (mingw32, mingw64, ucrt64)

### PR DESCRIPTION
**Pull request by mgondan**
This recipe is for a mingw32/mingw64/ucrt64 of the SWI-Prolog programming language. I made a second PR which is now based on a PKGBUILD template, as suggested in our previous conversation.

A few issues will be solved in future versions:

* a little patch that is needed to avoid overwriting dlls that are already installed on the MinGW system (e.g., zlib1.dll). I have already made the respective PR to SWI-Prologs development version.
* some of the installed dlls match the source folders. I have reported this potential problem to the SWI-Prolog developers.
* a few compiler warnings

Best regards,

Matthias

### CODE REVIEW

File: mingw-w64-swi-prolog/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+_realname=swi-prolog
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=8.4.2
+pkgrel=1
+pkgdesc="Prolog environment (mingw-w64)"
+arch=(any)
+mingw_arch=('mingw64' 'mingw32' 'ucrt64')
+url="https://www.swi-prolog.org/"
+license=("FreeBSD")
```
Comment (MehdiChinoune) [inline mingw-w64-swi-prolog/PKGBUILD:12]:
```suggestion
license=("FreeBSD")
depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
         "${MINGW_PACKAGE_PREFIX}-libwinpthread-git"
         "${MINGW_PACKAGE_PREFIX}-zlib")
```

File: mingw-w64-swi-prolog/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-make"
```
Comment (MehdiChinoune) [inline mingw-w64-swi-prolog/PKGBUILD:15]:
```suggestion
             "${MINGW_PACKAGE_PREFIX}-ninja"
```

File: mingw-w64-swi-prolog/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  declare -a extra_config
+  if check_option "debug" "n"; then
+    extra_config+=("-DCMAKE_BUILD_TYPE=Release")
+  else
+    extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  ${MINGW_PREFIX}/bin/cmake \
+    -G"MSYS Makefiles" \
```
Comment (MehdiChinoune) [inline mingw-w64-swi-prolog/PKGBUILD:45]:
```suggestion
    -G"Ninja" \
```

File: mingw-w64-swi-prolog/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+sha256sums=('be21bd3d6d1c9f3e9b0d8947ca6f3f5fd56922a3819cae03251728f3e1a6f389'
+  'ba4d9e55fd37ff4f540e4014a5e3b7c57c55fd2b51e74a481973c3d6f9e9f34b')
+
+prepare() {
+  cd ${srcdir}/swipl-${pkgver}
+  patch -p1 < ${srcdir}/01-skip-installation-of-system-dlls.patch
+}
+
+build() {
+  cd "${srcdir}/swipl-${pkgver}"
```
Comment (MehdiChinoune) [inline mingw-w64-swi-prolog/PKGBUILD:33]:
```suggestion
```

Comment (MehdiChinoune) [inline mingw-w64-swi-prolog/PKGBUILD:33]:
```suggestion
```

**Comment by lazka** [comment]
The directory structure is still a bit all over the place with generic names unlike the Linux package.. but this is a good start I guess.

**Comment by lazka** [comment]
thanks

**Pull request merged**